graph-data.rs/src/check_channels.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result as Fallible;
+use itertools::Itertools;
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+
+use cincinnati::plugins::internal::openshift_secondary_metadata_parser::plugin::graph_data_model::Channel;
+
+static CHANNEL_ORDER: [&'static str; 3] = ["stable", "fast", "candidate"];
+
+pub async fn run(channels_vec: &Vec<Channel>) -> Fallible<()> {
+    let mut errors: Vec<String> = vec![];
+
+    println!("Verifying all releases are following channel order");
+    // Make a hashmap of channels so that we could fetch a list of releases by name
+    let channels: HashMap<String, Vec<semver::Version>> = channels_vec
+        .iter()
+        .map(|c| (c.name.clone(), c.versions.clone()))
+        .collect();
+
+    // Collect a list of releases (4.6, 4.5 etc.)
+    let releases: Vec<&str> = channels_vec
+        .iter()
+        .filter_map(|c| c.name.split("-").last())
+        .unique()
+        .collect();
+
+    // Prepare a list of expected channels, ordered according to the rule
+    for r in releases {
+        let release_vec: Vec<String> = CHANNEL_ORDER
```

Comment:
Special-case 4.1, where we had `presubmit-4.1` and `stable-4.1`?

Reply:
I guess so. @vrutkovs can confirm.